services/job_chat/evaluation/test_prompts_issue_97_a.ipynb

```diff
@@ -0,0 +1,966 @@
+      "INFO:httpx:HTTP Request: POST https://api.anthropic.com/v1/messages?beta=prompt_caching \"HTTP/1.1 200 OK\"\n",
+      "What's the syntax to create a new ODK form submission using data from multiple CommCare cases?\n",
+      "INFO:httpx:HTTP Request: POST https://api.anthropic.com/v1/messages?beta=prompt_caching \"HTTP/1.1 200 OK\"\n",
+      "How do I transform a Google Sheets row into a CommCare case update?\n",
+      "INFO:httpx:HTTP Request: POST https://api.anthropic.com/v1/messages?beta=prompt_caching \"HTTP/1.1 200 OK\"\n",
+      "What's the best way to map nested CommCare form data to SatuSehat's FHIR resources?\n",
+      "INFO:httpx:HTTP Request: POST https://api.anthropic.com/v1/messages?beta=prompt_caching \"HTTP/1.1 200 OK\"\n",
+      "How can I merge data from multiple OpenMRS observations into a single Google Sheet row?\n",
+      "INFO:httpx:HTTP Request: POST https://api.anthropic.com/v1/messages?beta=prompt_caching \"HTTP/1.1 200 OK\"\n",
+      "What's the correct way to use each() to process multiple FHIR resources and create CommCare cases?\n",
+      "INFO:httpx:HTTP Request: POST https://api.anthropic.com/v1/messages?beta=prompt_caching \"HTTP/1.1 200 OK\"\n",
+      "How do I map an array of ODK repeat groups to create multiple OpenMRS encounters?\n",
+      "INFO:httpx:HTTP Request: POST https://api.anthropic.com/v1/messages?beta=prompt_caching \"HTTP/1.1 200 OK\"\n",
+      "What's the syntax for filtering Google Sheets data before sending it to SatuSehat?\n",
+      "INFO:httpx:HTTP Request: POST https://api.anthropic.com/v1/messages?beta=prompt_caching \"HTTP/1.1 200 OK\"\n",
+      "How can I use combine() to merge data from multiple sources before creating a FHIR resource?\n",
+      "INFO:httpx:HTTP Request: POST https://api.anthropic.com/v1/messages?beta=prompt_caching \"HTTP/1.1 200 OK\"\n",
+      "What's the proper way to map CommCare case properties to custom OpenMRS patient attributes?\n",
+      "INFO:httpx:HTTP Request: POST https://api.anthropic.com/v1/messages?beta=prompt_caching \"HTTP/1.1 200 OK\"\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Get answers to generated questions\n",
+    "qa_pairs = []\n",
+    "for question in generated_questions[:30]:\n",
+    "    answer = generate_apollo_format(question)\n",
+    "    qa_pairs.append((question, answer, \"v1\"))\n",
+    "for question in generated_questions[:30]: # two loops to make use of prompt caching\n",
+    "    new_answer = generate_new_format(question, system_prompt_v2)\n",
+    "    qa_pairs.append((question, new_answer, \"v2\"))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 337,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "evaluate_answer_template = \"\"\"\n",
+    "For the given Question and Answer pair, evaluate whether the answer at least briefly answers the question (True), or if it entirely fails to answer the question (False).\n",
```

Comment:
For your test, I would argue that we don;t need to check whether it answers the question. It's sufficient just to look for that nested state argument.

I'd also argue you can do this with regular python rather than using AI? But I'll leave that up to you